source/MicroBitAccelerometer.cpp

```diff
@@ -35,321 +34,84 @@ DEALINGS IN THE SOFTWARE.
 #include "LSM303Magnetometer.h"
 
 
-Accelerometer* MicroBitAccelerometer::driver;
+using namespace codal;
 
-MicroBitAccelerometer::MicroBitAccelerometer(MicroBitI2C &i2c, uint16_t id) : Accelerometer(coordinateSpace)
+Accelerometer* MicroBitAccelerometer::driver = NULL;
+
+
+MicroBitAccelerometer::MicroBitAccelerometer(MicroBitI2C &i2c, CoordinateSpace &coordinateSpace, uint16_t id) : Accelerometer(coordinateSpace, id)
 {
     autoDetect(i2c);
 }
 
-/**
- * Device autodetection. Scans the given I2C bus for supported accelerometer devices.
- * if found, constructs an appropriate driver and returns it.
- *
- * @param i2c the bus to scan. 
- * @param id the unique EventModel id of this component. Defaults to: MICROBIT_ID_ACCELEROMETER
- *
- */
+MicroBitAccelerometer::MicroBitAccelerometer(CoordinateSpace &coordinateSpace, uint16_t id) : Accelerometer(coordinateSpace, id)
+{
+}
+
 Accelerometer& MicroBitAccelerometer::autoDetect(MicroBitI2C &i2c)
 {
     static bool autoDetectCompleted = false;
     static CoordinateSpace coordinateSpace(SIMPLE_CARTESIAN, true, COORDINATE_SPACE_ROTATED_0);
     static NRF52Pin irq1(ID_PIN_IRQ1, P0_25, PIN_CAPABILITY_AD);
 
-    /*
-     * In essence, the LSM needs at least 6.4ms from power-up before we can use it.
-     * https://github.com/microbit-foundation/codal-microbit/issues/33
-     */
-    target_wait(10);
-
-    // Add pullup resisitor to IRQ line (it's floating ACTIVE LO)
-    irq1.getDigitalValue();
-    irq1.setPull(PullMode::Up);
-    irq1.setActiveLo();
```

Comment:
Moved inside the `if (!autoDetectCompleted)` to only wait and configure the GPIO once.